### PR TITLE
Speed up Aircraft.Repulse.

### PR DESCRIPTION
**Pull request by RoosterDragon**
Avoid LINQ in the repulsion force calculation, and cache a trait lookup used when determining movement speed.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Air/Aircraft.cs [hunk 1]
```diff
@@ -124,7 +125,11 @@ public Aircraft(ActorInitializer init, AircraftInfo info)
 			IsPlane = !info.CanHover;
 		}
 
-		public void Created(Actor self) { um = self.TraitOrDefault<UpgradeManager>(); }
+		public void Created(Actor self)
+		{
+			um = self.TraitOrDefault<UpgradeManager>();
+			speedModifiers = self.TraitsImplementing<ISpeedModifier>().ToArray().Select(sm => sm.GetSpeedModifier());
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: Won't this make it impossible for the speed modifiers to change over the course of this actor's life?

Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: Remember it's an enumerable, the `Select` will be re-evaluated every time.

Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: Oh right, carry on.

Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: The enumerable is passed directly into `ApplyPercentageModifiers` which does the heavy lifting.

So all we're doing is passing a field directly to a method - it doesn't come cheaper than that.

**Comment by pchote** [comment]
Looks good to me :+1:

File: OpenRA.Mods.Common/Traits/Air/Aircraft.cs (hunk 1 above)
Comment (penev92) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: I'm curious - is the .ToList() in the middle there really helping?

Comment (atlimit8) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: Yes, the `IEnumerable<int>` created by `.Select(sm => sm.GetSpeedModifier())` holds a reference to the unnamed `ISpeedModifier[]` created by `.ToArray()`.

Comment (penev92) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: Neat, thanks.

**Comment by atlimit8** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/453a36bfa96816ac224f45a6b3fae2fa3ec1eaae...81efcab0d7a701c6fe52c53fc7f87a4c0aeef542)

**Pull request merged**